vochain/scrutinizer/process.go

```diff
@@ -1,87 +1,139 @@
 
 	"go.vocdoni.io/dvote/log"
-	"go.vocdoni.io/dvote/types"
-	"go.vocdoni.io/dvote/util"
 )
 
 // ProcessInfo returns the available information regarding an election process id
-func (s *Scrutinizer) ProcessInfo(pid []byte) (*models.Process, error) {
-	return s.VochainState.Process(pid, false)
+func (s *Scrutinizer) ProcessInfo(pid []byte) (*Process, error) {
+	proc := &Process{}
+	err := s.db.FindOne(proc, badgerhold.Where("ID").Eq(pid))
+	return proc, err
 }
 
 // ProcessList returns the list of processes (finished or not) for a specific entity.
-func (s *Scrutinizer) ProcessList(entityID []byte, fromID []byte, max int64) ([][]byte, error) {
-	plistkey := []byte{types.ScrutinizerEntityPrefix}
-	plistkey = append(plistkey, entityID...)
-	processList, err := s.Storage.Get(plistkey)
-	if err != nil {
-		return nil, err
-	}
-	processListResult := [][]byte{}
-	fromLock := len(fromID) > 0
-	for _, process := range util.SplitBytes(processList, types.ProcessIDsize) {
-		if max < 1 || len(process) < 1 {
-			break
-		}
-		if !fromLock {
-			keyCopy := make([]byte, len(process))
-			copy(keyCopy, process)
-			processListResult = append(processListResult, keyCopy)
-			max--
-		}
-		if fromLock && bytes.Equal(fromID, process) {
-			fromLock = false
-		}
+func (s *Scrutinizer) ProcessList(
+	entityID []byte,
+	namespace uint32,
+	from,
+	max int) ([][]byte, error) {
+	procs := [][]byte{}
+	var err error
+
+	switch {
+	case namespace == 0 && len(entityID) > 0:
+		err = s.db.ForEach(
+			badgerhold.Where("EntityID").
+				Eq(entityID).
+				Index("EntityID").
+				SortBy("CreationTime").
+				Skip(from).
+				Limit(max),
+			func(p *Process) error {
+				procs = append(procs, p.ID)
+				return nil
+			})
+	case namespace > 0 && len(entityID) == 0:
+		err = s.db.ForEach(
+			badgerhold.Where("Namespace").
+				Eq(namespace).
+				Index("Namespace").
+				SortBy("CreationTime").
+				Skip(from).
+				Limit(max),
+			func(p *Process) error {
+				procs = append(procs, p.ID)
+				return nil
+			})
+	default:
+		err = s.db.ForEach(
+			badgerhold.Where("EntityID").
+				Eq(entityID).
+				And("Namespace").
+				Eq(namespace).
+				Index("EntityID").
+				SortBy("CreationTime").
+				Skip(from).
+				Limit(max),
+			func(p *Process) error {
+				procs = append(procs, p.ID)
+				return nil
+			})
 	}
-	return processListResult, nil
+
+	return procs, err
 }
 
-// ProcessListWithResults returns the list of process ID with already computed results
-func (s *Scrutinizer) ProcessListWithResults(max int64, fromID []byte) []string {
-	process := []string{}
-	for _, p := range s.List(max, fromID, []byte{types.ScrutinizerResultsPrefix}) {
-		process = append(process, fmt.Sprintf("%x", p))
-	}
-	return process
+// ProcessListWithResults returns the list of process ID with already computed results.
+// TODO: use [][]byte instead of []string as return value
+func (s *Scrutinizer) ProcessListWithResults(max, from int) []string {
+	procs := []string{}
+	_ = s.db.ForEach(
```

Comment:
ignoring the error altogether seems like a bad idea. at least you should log it.

Reply:
The reason i ignored it was because I'm returning "nil" on the ForEach callback. But looking deeper it could also return an error if the ForEach statement has a wrong type. Thus I'll capture the error, makes sense. Thanks.